proto/backup.proto

```diff
@@ -187,6 +188,23 @@ message GCS {
     string credentials_blob = 6;
 }
 
+// S3 and GCS can be transitioned to use this
+message Bucket {
+    string endpoint = 1;
+    string region = 3;
+    string bucket = 4;
+    string prefix = 5;
+    string storage_class = 6;
+}
+
+// CloudDynamic allows adding new fields without changing protobuf definitions
+message CloudDynamic {
+    Bucket bucket = 1;
+    string provider_name = 2;
+    map<string, string> attrs = 3;
```

Comment:
what is being held in the map? Can we define it as a struct (protobuf message) so that the fields are well defined?

Reply:
I added a comment about the purpose of CloudDynamic. The fields in Bucket are common, but otherwise every cloud provider can have arbitrary attributes.